R/utils.R

```diff
@@ -269,6 +269,50 @@ dirExists <- function(paths) {
   file.exists(paths) & file.info(paths)$isdir
 }
 
+# Removes empty directory (vectorized). This is needed because file.remove()
+# on Unix will remove empty directories, but on Windows, it will not. On
+# Windows, you would need to use unlink(recursive=TRUE), which is not very
+# safe. This function does it safely on Unix and Windows.
+dirRemove <- function(path) {
+  for (p in path) {
+    if (!dirExists(p)) {
+      stop("Cannot remove non-existent directory ", p, ".")
+    }
+    if (length(dir(p, all.files = TRUE, no.. = TRUE)) != 0) {
+      stop("Cannot remove non-empty directory ", p, ".")
+    }
+    result <- unlink(p, recursive = TRUE)
+    if (result == 1) {
+      stop("Error removing directory ", p, ".")
+    }
+  }
+}
+
+# If the path exists, this returns a canonical path (like normalizePath). If it
+# does not exist, it returns an absolute path, relative to the current dir. The
+# difference is that a canonical path follows symlinks and doesn't have any
+# `..`, while an absolute path here is simply one that starts with `/`.
+absolutePath <- function(path) {
```

Comment:
Since we need this path to exist anyway, if the dir doesn't exist we can create it and then call `normalizePath` on it.